llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -967,11 +1117,16 @@ void addInstrRequirements(const MachineInstr &MI,
   case SPIRV::OpConstantSampler:
     Reqs.addCapability(SPIRV::Capability::LiteralSampler);
     break;
+  case SPIRV::OpAccessChain:
```

Comment:
It's valid and makes sense now because you insert only `OpAccessChain` above in the code, so now you refer just to this single opcode `OpAccessChain`. My question is however, is it possible to produce a more general code in this place and refer to more than a single instruction `OpAccessChain`, meaning other instructions which have the same semantics (OpInBoundsAccessChain?..). Would it produce a better and more flexible piece of code, ready to any relevant changes of the Backend in future? I'd like to avoid a partial implementation when the SPIRV spec is satisfied in 50% -- for example, just for 1 of 2 possible opcodes satisfies the spec requirement wrt. required capabilities.

Reply:
Okay I added a case for OpInBoundsAccessChain. For now it the same as OpAccessChain. I generally don't think about OpInBoundsAccessChain because in HLSL we can't know that it is inbounds, so we don't generate the opcode. However, other languages might want to use it, so it is best to implement it.

However, I'm not sure how to test it because we do not generate it. I will rely on the testing for OpAccessChain.